pkg/controller/cloneset/cloneset_event_handler.go

```diff
@@ -161,6 +176,51 @@ func (e *podEventHandler) Update(evt event.UpdateEvent, q workqueue.RateLimiting
 	}
 }
 
+func (e *podEventHandler) shouldIgnoreUpdate(req *reconcile.Request, oldPod, curPod *v1.Pod) bool {
+	if curPod.Generation != oldPod.Generation {
+		return false
+	}
+
+	cs := &appsv1alpha1.CloneSet{}
+	if err := e.Get(context.TODO(), req.NamespacedName, cs); err != nil {
+		return false
+	}
+
+	if lifecycleFinalizerChanged(cs, oldPod, curPod) {
+		return false
+	}
+
+	coreControl := clonesetcore.New(cs)
+	if coreControl.IsPodUpdatePaused(oldPod) != coreControl.IsPodUpdatePaused(curPod) {
+		return false
+	}
+
+	containsReadinessGate := func(pod *v1.Pod) bool {
+		for _, r := range pod.Spec.ReadinessGates {
+			if r.ConditionType == appspub.InPlaceUpdateReady {
+				return true
+			}
+		}
+		return false
+	}
+
+	if containsReadinessGate(curPod) {
+		opts := coreControl.GetUpdateOptions()
+		opts = inplaceupdate.SetOptionsDefaults(opts)
+		if err := opts.CheckPodUpdateCompleted(curPod); err == nil {
+			if cond := inplaceupdate.GetCondition(curPod); cond == nil || cond.Status != v1.ConditionTrue {
```

Comment:
how about the case !opts.CheckPodUpdateCompleted(curPod) && inplaceupdate.GetCondition(curPod) == v1.ConditionTrue